### PR TITLE
Enable transport for host metadata calls

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Before this change, the agent did not use the default/global transport used in the rest of the agent. As a result host metadata calls would not respect the `no_proxy` list when configured in the `datadog.yaml`. 

This change uses the transport for all metadata calls as well as by default adds the metadata IP to the `no_proxy` list as to not break any existing users. This change also includes a new flag to disable this behavior and use the proxy for metadata calls. 

### Motivation

We have many reports of users expecting the config's `no_proxy` to work for host metadata calls. 

### Describe your test plan

Set an invalid proxy and watch the logs to see if it is attempting to use the proxy or not.

### CODE REVIEW

File: releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Use the default agent transport for host metadata calls.
+    This allowse usage of the config `no_proxy` setting for host metadata calls.
+    By default meatadata IP's are added to the transport's `no_proxy` list.
```
Comment (remeh) [inline releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml:13]: Nit: 1 typo each line

**Review by remeh — APPROVED** [review]
LGTM

File: pkg/config/config.go [hunk 2]
```diff
@@ -867,6 +867,11 @@ func loadProxyFromEnv(config Config) {
 		config.Set("proxy.no_proxy", p.NoProxy)
 		proxies = p
 	}
+
+	if !config.GetBool("proxy.use_proxy_for_host_metadata") {
```
Comment (hush-hush) [inline pkg/config/config.go:871]: Where is this defined in the config object ? We need to set a default and bind it to the environment.

File: releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Use the default agent transport for host metadata calls.
+    This allows usage of the config `no_proxy` setting for host metadata calls.
+    By default metadata IP's are added to the transport's `no_proxy` list.
```
Comment (hush-hush) [inline releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml:13]: I think we should make it clear that we're talking about cloud metadata and not datadog:

```suggestion
    By default cloud provider metadata IP's are added to the transport's `no_proxy` list.
```

WDYT ?

File: releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Added config flag `proxy.use_proxy_for_host_metadata` to disable this behavior.
```
Comment (hush-hush) [inline releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml:14]: Is there a use case where we would want to go through the proxy for this ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml (hunk 4 above)
Comment (gh123man) [reply releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml:14]: Unsure of a specific case. However the intention here is to allow the flexibility to use the proxy if someone wanted to.

File: pkg/config/config.go [hunk 5]
```diff
@@ -186,6 +186,7 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("disable_py3_validation", false)
 	config.BindEnvAndSetDefault("python_version", DefaultPython)
 	config.BindEnvAndSetDefault("allow_arbitrary_tags", false)
+	config.BindEnvAndSetDefault("use_proxy_for_cloud_metadata", false)
```
Comment (hush-hush) [inline pkg/config/config.go:189]: Since we talk about this in the changelog we should add it to the configuration template.

**Review by hush-hush — CHANGES_REQUESTED** [review]
One last comment

File: pkg/config/config.go (hunk 5 above)
Comment (gh123man) [reply pkg/config/config.go:189]: Updated - let me know if the wording sounds good to you.

File: pkg/config/config_template.yaml [hunk 6]
```diff
@@ -287,6 +287,12 @@ api_key:
 #   - "sensitive_key_1"
 #   - "sensitive_key_2"
 
+## @param use_proxy_for_cloud_metadata - bool - optional - default: false
```
Comment (kayayarai) [inline pkg/config/config_template.yaml:290]:
```suggestion
## @param use_proxy_for_cloud_metadata - boolean - optional - default: false
```

File: releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Use the default agent transport for host metadata calls.
+    This allows usage of the config `no_proxy` setting for host metadata calls.
+    By default cloud provider IP's are added to the transport's `no_proxy` list.
```
Comment (kayayarai) [inline releasenotes/notes/enable-transport-for-metadata-calls-727408e9f5916cb7.yaml:13]:
```suggestion
    By default cloud provider IPs are added to the transport's `no_proxy` list.
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
A couple minor edits, but otherwise lgtm from docs perspective

**Review by hush-hush — APPROVED** [review]

**Pull request merged**